frontend/src/components/custom-tools/output-for-doc-modal/OutputForDocModal.jsx

```diff
@@ -117,13 +118,63 @@ function OutputForDocModal({
 
   const updatePromptOutput = (data) => {
     setPromptOutputs((prev) => {
-      // If data is provided, use it; otherwise, create a copy of the previous state
-      const updatedPromptOutput = data || [...prev];
+      const updatedPromptOutput = getUpdatedPromptOutput(data, prev);
```

Comment:
@jagadeeswaran-zipstack why this changed to function call?

Reply:
Refactored code to reduce nesting.